App submission: Oak Node

ok300 opened the pull request:
# App Submission

### App name
Oak Node

### 256x256 SVG icon

https://oak-node.net/doc/tip/www/static/oak-logo.svg

### Gallery images
_(Upload 3 to 5 high-quality gallery images (1440x900px) of your app in PNG format, or just upload 3 to 5 screenshots of your app and we'll help you design the gallery images.)_

![Screenshot 2022-07-05 at 12-06-09 Oak](https://user-images.githubusercontent.com/106775972/177329403-deffb8fa-8c2d-47da-9043-cf86ea5e0d66.png)

![Screenshot 2022-07-05 at 14-32-38 Oak](https://user-images.githubusercontent.com/106775972/177329405-507f31cd-ee6e-476c-9767-e7223430e86f.png)

![Screenshot 2022-07-05 at 12-05-52 Oak](https://user-images.githubusercontent.com/106775972/177329400-dc61bd70-acb6-40c8-8eb1-f6d1634de970.png)


### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

---

Looking forward to your feedback. Thanks!

nevets963 reviewed (approved):
Hey @ok300 Thanks a lot for your app submission! It's really impressive to see pretty much a perfect PR considering we changed the app structure recently! Awesome work!

I've tested the app on both AMD64 and ARM64 (Pi) and they're working perfectly.

1 thing I noted on my regtest node is this tiny error by 'Next Run':
<img width="1063" alt="Screenshot 2022-07-05 at 22 14 22" src="https://user-images.githubusercontent.com/1112574/177365917-054f5920-62df-4c5b-a301-4335276ca8b7.png">

I can see why and I don't think it's a blocking issue; just an observation.

I also noted that you're not caching any of the static assets (css, js, etc files). I think this is OK for local network access, but if the app is accessed over Tor (which is really slow) or Tailscale, then the latency could have some impact. Of course, I understand you then need to think about cache invalidation when you update these assets, but something to think about. I won't block the PR on that either...

I would also consider a much punchy description e.g.
```
Oak Node is a self-hosted a Scheduled Payments module built for LND that lets you support your favorite content creators by regularly sending them tips to their Lightning Address.

It also includes an optional bot module, which lets you interact with your Oak Node from a chat interface.
```
But again, I won't block on that, your call :)

This is approved from me!

@mayankchhabra Could you help with preparing assets for the App Store?

@lukechilds Could you also test Oak Node and check/fix up the container IP once you're happy to merge?

ok300 commented:
@nevets963 : thanks for the kind words and detailed feedback :)

About the things you found:

- the "Next Run" issue you saw on regtest: I've seen it in some tests where LND is not fully up and running. Not a bug (doesn't affect functionality) but yes it is annoying. Added to the list, will address it in the next release.

- the caching of static assets: unfortunately its a limitation of a library I'm using. As soon as they add support for that, I'll integrate it.

As for the description, you're right it needed some polishing. I already updated it to be shorter and crispier.

Any more feedback is always welcome :)

ok300 commented:
FYI: I've bumped the app version, which fixes the 2 minor issues we discussed above.

Tested it on an Umbrel VM.

lukechilds commented on oak-node/exports.sh:
```diff
@@ -0,0 +1,2 @@
+export APP_OAK_NODE_IP="10.21.21.99"
```
```suggestion
export APP_OAK_NODE_IP="10.21.21.105"
```

lukechilds reviewed (approved):
Tested and working great!

lukechilds commented:
Nice work @ok300! This is ready to ship.

@mayankchhabra are you able to put together some release assets?

ok300 commented:
No worries @lukechilds, thanks for the review and IP fixes.

mayankchhabra commented:
Really nice work on Oak Node (and packaging it!) @ok300. The app store gallery images and app icon are now live via [getumbrel/umbrel-apps-gallery](https://github.com/getumbrel/umbrel-apps-gallery/tree/master/oak-node).

Good luck with the launch in the app store! :)

PS: We added a splash of color to the app icon. In case you'd like us to revert it, just let us know. And also if you need to update the gallery images or the icon in the future, feel free to submit a PR to the above repo. Cheers!

lukechilds commented:
Woop it's all live. 🚀

Congrats @ok300, Oak Node should propagate to all Umbrel app stores within the next 15 minutes.

ok300 commented:
Thank you both! I can see it in the appstore, and yes the logo did need a splash of color :)

plazareff commented:
Thanks for the reply @nmfretz !

It's true I had not double checked for Carlos' url as I was using the integrated support/contact form, I did not imagine it could fail. My prejudice ;-)

For the rest, there are 2 issues I can identify after some more investigation:

For the double ? it is because one is inside the callback URI given to Oak by the LNBit Pay Link extension which is like `/lnurlp/api/v1/lnurl/cb/5yZENh?webhook_data=alias%3D%27webhook_data%27+extra%3D%7B%7D`. Oak does not check if there is one ? or not and adds ?amount=x where, if it checked, that should be &amount=x

I could not get the Pay Link extension not to generate the webhook_data part of the callback url.

But there is another issue with services like getalby which gives callback urls like https://getalby.com/lnurlp/user/callback, so no double ? issue there. But for those, I get the error triggered by `ln_sdk::lnurl INFO: [lnurl-pay 4/5] Attempting to pay it`

So it maybe is not Oak but ln_sdk which I see is called in src/lnd.rs:
`use ln_sdk::client_v2::lnd::{LndClient, LndClientOps, LndConfig};`

For the extra ? I could trace the issue:
https://oak-node.net/cgit/ln-sdk.git/tree/src/lnurl.rs

`let mut invoice_lookup_resp = format!(
        "{}?amount={amount_msats}",
        onetime_pay_request.request.callback
    );`

For the Failed to call LND send_payment_sync error I receive with GetAlby, I'm afraid this is beyond my competence.

If you're in touch with Carlos, it may be worth updating his ln-sdk ?
